### ISSUE TITLE
add keywords to metadata

### ISSUE THREAD

**Issue by lacava** [EpistasisLab/pmlb#52]
It would be useful to tag metadata with keywords like "bioinformatics", "economics", "image", etc. I'm not sure of a principled way to do this, but I imagine we might want to use a reference ontology (like [ACM CSS?](https://dl.acm.org/ccs)). Discussion welcome.

**Comment by trangdata** [comment]
I'm in favor of keywords but I'm not sure ACM CSS is an ideal ontology for datasets.

**Comment by lacava** [comment]
Any idea what keyword ontologies are out there? I'm having trouble finding examples.

**Comment by lacava** [comment]
added in PR #69 :)

**Comment by trangdata** [comment]
Hmm... I thought we added a `keywords` field to all auto-generated metadata.yaml files that haven't been manually modified.

But I don't see `keywords`, e.g., [here](https://github.com/EpistasisLab/penn-ml-benchmarks/blob/master/datasets/210_cloud/metadata.yaml).

**Comment by lacava** [comment]
we haven't added keywords by default except to the [template](https://github.com/EpistasisLab/penn-ml-benchmarks/blob/master/metadata_template.yaml). i'm not sure how we would put them in auto-generated files? just have a blank entry?

**Comment by trangdata** [comment]
Oh I see. Hmm I was thinking like what you have in the template:
```
keywords:
  - keyword1
  - keyword2
```

**Comment by lacava** [comment]
i don't think we should have dummy keywords appearing in the auto-generated files

**Comment by trangdata** [comment]
Perhaps `keyword1` and `keyword2` are not great place holders. Maybe `needs_keyword` or `replace_me` would work? 

I think we need to make it as easy as possible for a contributor to come in with little knowledge of yaml and can still contribute. The contributor shouldn't have to come back to the template to find out what other fields are there to add. This will make the reviewer's job easier as well. I myself have missed this field most times when reviewing metadata files.

And having two place holders will also avoid a situation where keywords are put in as 
```
keywords:
  - keyword1, keyword2
```

I'm happy to implement this. I feel like it's a case of [practicality beats purity](https://www.python.org/dev/peps/pep-0020/) we have here.

**Comment by lacava** [comment]
How about we add a line like 

```yaml
keywords: # descriptive terms for the dataset, each hyphenated on a new line (see metadata_template.yaml)
```

EDIT: I think we can use `null` to be read in as [None in python](https://stackoverflow.com/questions/30134110/how-can-i-output-blank-value-in-python-yaml-file) 

```yaml
keywords: # descriptive terms for the dataset, each hyphenated on a new line (see metadata_template.yaml)
  - null # replace with keyword
```

**Comment by trangdata** [comment]
Addressed in #107.
